infrastructures/kubernetes/src/main/java/org/eclipse/che/workspace/infrastructure/kubernetes/wsplugins/brokerphases/ListenBrokerEvents.java

```diff
@@ -45,20 +48,26 @@ public ListenBrokerEvents(
       String workspaceId,
       KubernetesPluginsToolingValidator pluginsValidator,
       BrokersResult brokersResult,
-      EventService eventService) {
+      EventService eventService,
+      @Nullable Tracer tracer) {
     this.workspaceId = workspaceId;
     this.pluginsValidator = pluginsValidator;
     this.brokersResult = brokersResult;
     this.eventService = eventService;
+    this.tracer = tracer;
+    this.spanName = SPAN_NAME;
   }
 
+  @Override
   public List<ChePlugin> execute() throws InfrastructureException {
+    Span tracingSpan = startTracingPhase();
     BrokerStatusListener brokerStatusListener =
         new BrokerStatusListener(workspaceId, pluginsValidator, brokersResult);
     try {
       LOG.debug("Subscribing broker events listener for workspace '{}'", workspaceId);
       eventService.subscribe(brokerStatusListener, BrokerEvent.class);
 
+      finishSpanIfExists(tracingSpan);
```

Comment:
Subscribing is immediate whereas listening is not. But actual listening is not traced here. Should we remove this tracing?

Reply:
You're right, the goal was to represent all phases of brokering, but listen events is a >1ms operation. Removed.